docs/pages/guides/hello-world/browser-wallet.mdx

```diff
@@ -0,0 +1,224 @@
+
+  const publicClient = createPublicClient(clientOptions);
+
+  /*
+   * Create an observable for contract writes that we can
+   * pass into MUD dev tools for transaction observability.
+   */
+  const write$ = new Subject<ContractWrite>();
+
+  /*
+   * Create a temporary wallet and a viem client for it
+   * (see https://viem.sh/docs/clients/wallet.html).
+   */
+  /*
+  const burnerAccount = createBurnerAccount(networkConfig.privateKey as Hex);
+  
+  const burnerWalletClient = createWalletClient({
+    ...clientOptions,
+    account: burnerAccount,
+  })
+    .extend(transactionQueue())
+    .extend(writeObserver({ onWrite: (write) => write$.next(write) }));
+  */
+
+  const [address] = await window.ethereum.request({
+    method: "eth_requestAccounts",
+  });
+
+  const extensionWalletClient = createWalletClient({
+    ...clientOptions,
+    transport: custom(window.ethereum!),
+    account: address,
+  });
+
+  const chainId = await extensionWalletClient.getChainId();
+  if (chainId != networkConfig.chain.id) await extensionWalletClient.addChain({ chain: networkConfig.chain });
+
+  /*
+   * Create an object for communicating with the deployed World.
+   */
+  const worldContract = getContract({
+    address: networkConfig.worldAddress as Hex,
+    abi: IWorldAbi,
+    client: { public: publicClient, wallet: extensionWalletClient /* burnerWalletClient */ },
+  });
+
+  /*
+   * Sync on-chain state into RECS and keeps our client in sync.
+   * Uses the MUD indexer if available, otherwise falls back
+   * to the viem publicClient to make RPC calls to fetch MUD
+   * events from the chain.
+   */
+  const { components, latestBlock$, storedBlockLogs$, waitForTransaction } = await syncToRecs({
+    world,
+    config: mudConfig,
+    address: networkConfig.worldAddress as Hex,
+    publicClient,
+    startBlock: BigInt(networkConfig.initialBlockNumber),
+  });
+
+  return {
+    world,
+    components,
+    playerEntity: encodeEntity({ address: "address" }, { address: extensionWalletClient.account.address }),
+    publicClient,
+    walletClient: extensionWalletClient /* burnerWalletClient, */,
+    latestBlock$,
+    storedBlockLogs$,
+    waitForTransaction,
+    worldContract,
+    write$: write$.asObservable().pipe(share()),
+  };
+}
+```
+
+</CollapseCode>
+
+<details>
+
+<summary>Explanation</summary>
+
+```ts
+import {
+    .
+    .
+    .
+  custom,
+} from "viem";
+
+```
+
+To use a browser wallet, you need to use a [`custom`](https://viem.sh/docs/clients/transports/custom) transport.
+
+```ts
+const [address] = await window.ethereum.request({
+  method: "eth_requestAccounts",
+});
+```
+
+Request the [address list](https://viem.sh/docs/accounts/jsonRpc) from the browser extension (confusingly, also called "wallet").
+This request prompts the extension to ask for user permission to connect to the application.
+
+```ts
+const extensionWalletClient = createWalletClient({
+  ...clientOptions,
+  transport: custom(window.ethereum!),
```

Comment:
I don't think this is a pattern we should support. A complete browser wallet solution will come with Account Kit (name TBD).

Reply:
I think we should explain all the available options (Account Kit when it's available) and let developers decide what's best for them. 

@alvrs , can you tie break and decide?

Reply:
We're gonna prioritize shipping a usable (and documentable) version of the account kit libraries very soon so I would agree with @holic here that we shouldn't merge this one yet so we don't send folks off a path they have to walk back later. (Note that the account kit libraries don't require you to use all of account kit, but also just make setting up an external wallet and delegations to a local signer much easier.)

If we extract this change to its own PR you can still send the WIP to folks asking about external wallets in the discord, with the disclaimer that a much better solution is in the works and ready soon.

Reply:
Removed here, added as https://github.com/latticexyz/mud/pull/2924